tensorflow_addons/callbacks/tests/avg_model_checkpoint_test.py

```diff
@@ -0,0 +1,263 @@
+        batch_size=BATCH_SIZE,
+        validation_data=(x, y),
+        callbacks=[avg_model_ckpt],
+    )
+    assert not os.path.exists(test_model_filepath)
+
+
+def test_save_freq(tmp_path):
+    test_filepath = str(tmp_path / "test_model.{epoch:02d}.h5")
+    x, y, model = get_data_and_model()
+    save_freq = "epoch"
+    avg_model_ckpt = AverageModelCheckpoint(
+        update_weights=False, filepath=test_filepath, save_freq=save_freq
+    )
+    model.fit(
+        x,
+        y,
+        epochs=EPOCHS,
+        batch_size=BATCH_SIZE,
+        validation_data=(x, y),
+        callbacks=[avg_model_ckpt],
+    )
+    assert os.path.exists(test_filepath.format(epoch=1))
+    assert os.path.exists(test_filepath.format(epoch=2))
+    assert os.path.exists(test_filepath.format(epoch=3))
+    assert os.path.exists(test_filepath.format(epoch=4))
+    assert os.path.exists(test_filepath.format(epoch=5))
+
+
+def test_invalid_save_freq(tmp_path):
+    test_model_filepath = str(tmp_path / "test_model.h5")
+    save_freq = "invalid_save_freq"
+    with pytest.raises(ValueError, match="Unrecognized save_freq"):
+        AverageModelCheckpoint(
+            update_weights=True, filepath=test_model_filepath, save_freq=save_freq
+        )
+
+
+def _get_dummy_resource_for_checkpoint_testing(tmp_path):
+    def get_input_datasets():
+        # Simple training input.
+        train_input = [[1.0]] * 16
+        train_label = [[0.0]] * 16
+        ds = tf.data.Dataset.from_tensor_slices((train_input, train_label))
+        return ds.batch(8, drop_remainder=True)
+
+    # Very simple model to eliminate randomness.
+    optimizer = MovingAverage(
+        tf.keras.optimizers.SGD(lr=2.0), sequential_update=True, average_decay=0.5
+    )
+    model = keras.Sequential()
+    model.add(keras.layers.Dense(units=1, input_shape=(1,)))
+    model.compile(loss="mae", optimizer=optimizer, metrics=["mae"])
+    train_ds = get_input_datasets()
+    filepath = str(tmp_path / "test_model.{epoch:02d}.h5")
+    callback = AverageModelCheckpoint(
+        update_weights=False, filepath=filepath, save_weights_only=True
+    )
+    return model, train_ds, callback, filepath
+
+
+def _run_load_weights_on_restart_test_common_iterations():
+    (model, train_ds, callback, filepath) = _get_dummy_resource_for_checkpoint_testing()
+    initial_epochs = 3
+    model.fit(train_ds, epochs=initial_epochs, callbacks=[callback])
+    # The files should exist after fitting with callback.
+    for epoch in range(initial_epochs):
+        assert os.path.exists(filepath.format(epoch=epoch + 1))
+    model.fit(train_ds, epochs=1)
+    weights_after_one_more_epoch = model.get_weights()
+    # The filepath should continue to exist after fitting without callback.
+    for epoch in range(initial_epochs):
+        assert os.path.exists(filepath.format(epoch=epoch + 1))
+    return model, train_ds, filepath, weights_after_one_more_epoch
+
+
+def get_checkpoint_load_weights_on_restart_true_test():
```

Comment:
test functions should have this form: `test_*`. Otherwise they're not run.

Reply:
Sorry, didn't notice while naming.